packages/react-utils/src/components/FhirSelect/index.tsx

```diff
@@ -0,0 +1,169 @@
+          resourceType,
+          { page: pageParam, pageSize, search: debouncedSearchValue ?? null },
+          extraQueryParams
+        );
+        return response;
+      },
+      getNextPageParam: (lastGroup: IBundle, allGroups: IBundle[]) => {
+        const totalFetched = getTotalRecordsInBundles(allGroups);
+        const total = lastGroup.total as number;
+        if (totalFetched < total) {
+          return page + 1;
+        } else {
+          return false;
+        }
+      },
+      getPreviousPageParam: () => {
+        if (page === 1) {
+          return undefined;
+        } else {
+          return page - 1;
+        }
+      },
+      refetchOnWindowFocus: false,
+    });
+
+  const changeHandler = (
+    value: string,
+    fullOption: SelectOption<ResourceT> | SelectOption<ResourceT>[]
+  ) => {
+    const saneFullOption = Array.isArray(fullOption) ? fullOption.slice() : fullOption;
+    props.onChange?.(value, saneFullOption);
+    getFullOptionOnChange?.(saneFullOption);
+  };
+
+  const searchHandler = (value: string) => {
+    setSearchValue(value);
+  };
+
+  const options = ((data?.pages ?? []) as IBundle[]).flatMap((resourceBundle: IBundle) => {
+    const resources = getResourcesFromBundle<ResourceT>(resourceBundle);
+    const allOptions = resources.map(transformOption);
+    const saneOptions = allOptions.filter((option) => option !== undefined);
+    return saneOptions as SelectOption<ResourceT>[];
+  });
+
+  const pages = (data?.pages ?? []) as IBundle[];
+  const recordsFetchedNum = getTotalRecordsInBundles(pages);
+  const totalPossibleRecords = getTotalRecordsOnApi(pages);
+  const remainingRecords = totalPossibleRecords - recordsFetchedNum;
+
+  const propsToSelect = {
+    style: { minWidth: '200px' },
+    ...restProps,
+    onChange: changeHandler,
+    loading: isLoading,
+    notFoundContent: isLoading ? <Spin size="small" /> : <Empty description={t('No data')} />,
+    filterOption: false,
+    options: options,
+    searchValue,
+    dropdownRender: (menu: React.ReactNode) => (
+      <>
+        {!error && data && menu}
+        <Divider style={{ margin: '8px 0' }} />
+        {error ? (
+          <Alert message={t('Unable to load dropdown options.')} type="error" showIcon />
+        ) : (
+          <Space direction="vertical">
+            {data && (
+              <small style={{ padding: '4px 16px' }}>
+                {t('Showing {{recordsFetchedNum}}; {{remainingRecords}} more records.', {
+                  recordsFetchedNum,
+                  remainingRecords,
+                })}
+              </small>
+            )}
+            <Button
+              type="text"
+              icon={<PlusOutlined />}
+              disabled={!hasNextPage || isFetchingNextPage || isFetching}
+              loading={isFetchingNextPage}
+              onClick={() => fetchNextPage()}
+            >
+              {isFetchingNextPage ? t('Fetching next page') : t('Load more options')}
```

Comment:
I'd prefer if we used a caret icon (`>>`) here for showing load more options. Also showing how many more records we have; separate them from records fetched

Reply:
is this what you mean: 
![Uploading image.png…]()

Reply:
This image didn't upload

Reply:
![Screenshot from 2023-02-01 21-28-39](https://user-images.githubusercontent.com/28119869/216910024-d3d721e7-bf84-41a0-984d-e40185e2845c.png)

Reply:
@p-netm this does not have more records so no caret? I meant something like below:

Reply:
@machariamuguku so you mean instead of the button text reading: `+ load more options` it should read `load more options >>`

Reply:
@p-netm yes, but with the carets facing down. We can try a number of variations and go with the best? Maybe even with carets below like above block